src/main/scala/viper/gobra/frontend/ParseTreeTranslator.scala

```diff
@@ -406,16 +427,24 @@ class ParseTreeTranslator(pom: PositionManager, source: Source, specOnly : Boole
     * {@link #visitChildren} on {@code ctx}.</p>
     */
   override def visitInterfaceType(ctx: GobraParser.InterfaceTypeContext): PInterfaceType = {
-    val methodDecls = visitListNode[PMethodSig](ctx.methodSpec())
-    val embedded = visitListNode[PTypeName](ctx.typeName()).map {
-      case tn: PUnqualifiedTypeName => PInterfaceName(tn).at(ctx)
-      case _: PDot => fail(ctx, "Imported types are not yet supported as embedded fields.")
-      case _ => fail(ctx, s"Interface embeds predeclared type.")
-    }
-    val predicateDecls = visitListNode[PMPredicateSig](ctx.predicateSpec())
+    val methodDecls = visitNodeIf[PMethodSig, InterfaceElemContext](ctx.interfaceElem(), ctx => has(ctx.methodSpec()))
```

Comment:
why does this requires `visitNodeIf` now?

Reply:
We have one additional level of indirection now. We have to filter the interface element by their category (methodSpec, typeElem and predicateSpec).

Reply:
You can look at the cahnges of the interfaceType rule in the GobraParser